File: RgbLedHost/src/RgbLedHost/SSwVBHost/SSwController.vb
Imports Netduino.Foundation.LEDs
Imports System
Imports System.Collections

Namespace SSwVBHost
    Public Class SSwController
        Protected _rgbPwmLed As RgbPwmLed

        Public Sub New(ByVal rgbPwmLed As RgbPwmLed)
            _rgbPwmLed = rgbPwmLed
            _rgbPwmLed.SetColor(Netduino.Foundation.Color.Red)
        End Sub

        Public Sub TurnOn()
            _rgbPwmLed.[Stop]()
            _rgbPwmLed.SetColor(GetRandomColor())
        End Sub

        Public Sub TurnOff()
            _rgbPwmLed.[Stop]()
            _rgbPwmLed.SetColor(Netduino.Foundation.Color.FromHsba(0, 0, 0))
        End Sub

        Public Sub StartBlink()
            _rgbPwmLed.[Stop]()
            _rgbPwmLed.StartBlink(GetRandomColor())
        End Sub

        Public Sub StartPulse()
            _rgbPwmLed.[Stop]()
            _rgbPwmLed.StartPulse(GetRandomColor())
        End Sub

        Public Sub StartRunningColors()
            Dim arrayColors = New ArrayList()
            Dim i As Integer = 0

            While i < 360
                Dim hue = (CDbl(i) / 360.0F)
                arrayColors.Add(Netduino.Foundation.Color.FromHsba((CDbl(i) / 360.0F), 1, 1))
                i = i + 5
            End While

            Dim intervals As Integer() = New Integer(arrayColors.Count - 1) {}

            For i = 0 To intervals.Length - 1
                intervals(i) = 100
            Next

            _rgbPwmLed.[Stop]()
            _rgbPwmLed.StartRunningColors(arrayColors, intervals)
        End Sub

        Public Sub NetworkConnected()
            _rgbPwmLed.[Stop]()
            _rgbPwmLed.SetColor(Netduino.Foundation.Color.Green)
        End Sub

        Protected Function GetRandomColor() As Netduino.Foundation.Color
            Dim random = New Random()
            Return Netduino.Foundation.Color.FromHsba(random.NextDouble(), 1, 1)
        End Function
    End Class
End Namespace

File: SSwHost/SSwController.cs
using Microsoft.SPOT.Hardware;
using Netduino.Foundation.LEDs;
using SecretLabs.NETMF.Hardware.Netduino;
using System;
using System.Collections;

namespace SSwHost
{
    public class SSwController
    {
        // create red-green-blue(rgb) pulse width modulator(Pwm) output to an LED
        protected RgbPwmLed _rgbPwmLed;

        // create a relay output port (a port that can be written to) and connect it to Digital Pin 2
        protected OutputPort _rec1Relay = new OutputPort(Pins.GPIO_PIN_D2, false);

        // constructor that set up the initial state of the outputs
        public SSwController(RgbPwmLed rgbPwmLed)
        {
            _rgbPwmLed = rgbPwmLed;
            _rgbPwmLed.SetColor(Netduino.Foundation.Color.Red);
            _rec1Relay.Write(false);
        }

        // Turn on the receptacle 1 relay
        public void TurnOn1()
        {
            _rec1Relay.Write(true);
        }

        // Turn off the receptacle 1 relay
        public void TurnOff1()
        {
            _rec1Relay.Write(false);
        }

        // Feedback for the network connectivity
        public void StartBlink()
        {
            _rgbPwmLed.Stop();
            _rgbPwmLed.StartBlink(GetRandomColor());
        }

        public void StartPulse()
        {
            _rgbPwmLed.Stop();
            _rgbPwmLed.StartPulse(GetRandomColor());
        }

        public void StartRunningColors()
        {
            var arrayColors = new ArrayList();
            for (int i = 0; i < 360; i = i + 5)
            {
                var hue = ((double)i / 360F);
                arrayColors.Add(Netduino.Foundation.Color.FromHsba(((double)i / 360F), 1, 1));
            }

            int[] intervals = new int[arrayColors.Count];
            for (int i = 0; i < intervals.Length; i++)
            {
                intervals[i] = 100;
            }

            _rgbPwmLed.Stop();
            _rgbPwmLed.StartRunningColors(arrayColors, intervals);
        }

        public void NetworkConnected()
        {
            _rgbPwmLed.Stop();
            _rgbPwmLed.SetColor(Netduino.Foundation.Color.Green);
        }

        protected Netduino.Foundation.Color GetRandomColor()
        {
            var random = new Random();
            return Netduino.Foundation.Color.FromHsba(random.NextDouble(), 1, 1);
        }
    }
}